core/controllers/admin_test.py

```diff
@@ -1739,3 +1739,63 @@ def test_check_community_reviewer_rights_without_username(self):
 
         self.assertEqual(response['error'], 'Missing username param')
         self.logout()
+
+
+class IntereactionByExplorationIdHandlerTests(test_utils.GenericTestBase):
+    """Tests for interaction by exploration handler."""
+
+    EXP_ID = 'exp'
+
+    def setUp(self):
+        """Complete the signup process for self.ADMIN_EMAIL."""
+        super(IntereactionByExplorationIdHandlerTests, self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.editor_id = self.get_user_id_from_email(self.EDITOR_EMAIL)
+        self.exploration = self.save_new_valid_exploration(
+            self.EXP_ID, self.editor_id, end_state_name='End')
+
+    def test_interaction_by_exploration_id_handler(self):
+        self.login(self.ADMIN_EMAIL, is_super_admin=True)
+
+        # Test that it returns all interaction ids.
+        payload = {
+            'exploration_id': self.EXP_ID
+        }
+
+        response = self.get_json(
+            '/interactionsbyexplorationid', params=payload)
+        interactions_list = response['interactions']
+        self.assertEqual(
+            sorted(interactions_list), ['EndExploration', 'TextInput'])
+
+    def test_handler_with_invalid_exploration_id_raise_error(self):
```

Comment:
I think it might also be worthwhile to test two more cases:
1. An exploration with no states (should provide empty list of interaction IDs). This tests when the for loop yields nothing.
2. An exploration with some states with interaction IDs, and others without. This tests the interaction ID None check.

Reply:
For first one we need to setup a demo exploration in order to yield empty list